### PR TITLE
Add custom rosdoc2 config for ros2_control metapackage

### PR DESCRIPTION
**Pull request by christophfroehlich**
rosdoc2 will be used by the build farm to publish API docs on http://docs.ros.org/en/rolling/p/ros2_control/ which is empty now.
- remove doxyfile because it is added to control.ros.org and used here https://github.com/ros-controls/control.ros.org/pull/278
- Add custom landing page for rosdoc2: logo, links to control.ros.org
- add rosdoc2 workflow, but only if the relevant files changed

image

### CODE REVIEW

File: .github/workflows/rosdoc2.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  pull_request:
+    paths:
+      - ros2_control/doc/**
+      - ros2_control/rosdoc2.yaml
+      - ros2_control/package.xml
+
+
+jobs:
+  check:
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-rosdoc2.yml@rosdoc2
```
Comment (christophfroehlich) [inline .github/workflows/rosdoc2.yml:14]:
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-rosdoc2.yml@master
```

File: ros2_control/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    - `ROS Index <https://index.ros.org/p/controller_manager_msgs/>`__
+  * - hardware_interface
+    - `API <http://docs.ros.org/en/rolling/p/hardware_interface/>`__
+    - `ROS Index <https://index.ros.org/p/hardware_interface/>`__
+  * - ros2_control_test_assets
+    - `API <http://docs.ros.org/en/rolling/p/ros2_control_test_assets/>`__
+    - `ROS Index <https://index.ros.org/p/ros2_control_test_assets/>`__
+  * - transmission_interface
+    - `API <http://docs.ros.org/en/rolling/p/transmission_interface/>`__
+    - `ROS Index <https://index.ros.org/p/transmission_interface/>`__
```
[the remark this answers is not in the document]
Comment (christophfroehlich) [reply ros2_control/doc/index.rst]: yes!

File: ros2_control/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    - `API <http://docs.ros.org/en/rolling/p/hardware_interface/>`__
+    - `ROS Index <https://index.ros.org/p/hardware_interface/#rolling>`__
+  * - joint_limits
+    - `API <http://docs.ros.org/en/rolling/p/joint_limits/>`__
+    - `ROS Index <https://index.ros.org/p/joint_limits/#rolling>`__
+  * - ros2_control_test_assets
+    - `API <http://docs.ros.org/en/rolling/p/ros2_control_test_assets/>`__
+    - `ROS Index <https://index.ros.org/p/ros2_control_test_assets/#rolling>`__
+  * - transmission_interface
+    - `API <http://docs.ros.org/en/rolling/p/transmission_interface/>`__
```
Comment (saikishor) [inline ros2_control/doc/index.rst:31]: @christophfroehlich sorry for not writing earlier 

I see it's missing ros2controlcli, rqt_controller_manager etc

Comment (christophfroehlich) [reply ros2_control/doc/index.rst:31]: is this really helpful? I'm not sure, this is not meant to be of public API?
http://docs.ros.org/en/rolling/p/ros2controlcli/
http://docs.ros.org/en/rolling/p/rqt_controller_manager/

**Review by bmagyar — APPROVED** [review]
Thank you!

**Pull request merged**